### PR TITLE
Feature/multiline annotations

### PR DESCRIPTION
**Pull request by numcys**
### What is the feature/fix?
Added Support for adding multiline annotations to deployments. 
example manifest:
```
services:
  web:
    build: .
    command: node app.js
    port: 3000
    annotations:
        - "key1=value1"
        - key2: "value2"
        - key3: |
            this is it
        - ad.datadoghq.com/test.checks: |
            {   
                "my-test": {
                "init_config": {
                    "is_jmx": true,
                    "collect_default_metrics": true
                },
                "instances": [{
                    "host": "localhost",
                    "port": "1234"
                }]
            }
            }
```

### CODE REVIEW

File: pkg/common/format.go [hunk 1]
```diff
@@ -12,6 +13,10 @@ var (
 )
 
 func FormatYAML(data []byte) ([]byte, error) {
+
+	// Unescape HTML entities first
+	data = []byte(html.UnescapeString(string(data)))
```
Comment (nightfury1204) [inline pkg/common/format.go:18]: Why this is needed here?

File: pkg/manifest/service.go [hunk 2]
```diff
@@ -54,6 +55,13 @@ type Service struct {
 	AccessControl      AccessControlOptions  `yaml:"accessControl,omitempty"`
 }
 
+type Annotations []Annotation
+
+type Annotation struct {
+	Key   string      `yaml:"key"`
+	Value interface{} `yaml:"value"`
```
Comment (nightfury1204) [inline pkg/manifest/service.go:62]: why this is interface instead of string?

File: pkg/manifest/service.go [hunk 3]
```diff
@@ -335,18 +343,36 @@ func (sr ServiceResource) GetConfigMapKey() string {
...
-		annotations[parts[0]] = parts[1]
+		jsonStr := string(jsonValue)
+
+		if containsJSONStructure(jsonStr) {
+			key := fmt.Sprintf("%s : |", annotation.Key)
+			annotations[key] = jsonStr
+		} else {
+			key := fmt.Sprintf("%s :", annotation.Key)
+			annotations[key] = jsonStr
+		}
```
Comment (nightfury1204) [inline pkg/manifest/service.go:366]: why json marshal and json str here?

File: pkg/manifest/yaml.go [hunk 4]
```diff
@@ -158,6 +159,75 @@ func (v *Service) SetName(name string) error {
...
+						// Successfully unmarshaled JSON string into a map
+						annotations = append(annotations, Annotation{Key: keyStr, Value: jsonValue})
+					} else {
+						// If not a valid JSON, store the string as-is
+						annotations = append(annotations, Annotation{Key: keyStr, Value: jsonStr})
+					}
+				} else {
+					// Store as is if not a JSON string
+					annotations = append(annotations, Annotation{Key: keyStr, Value: v})
+				}
```
Comment (nightfury1204) [inline pkg/manifest/yaml.go:200]: why json unmarshal here?

File: provider/k8s/template.go [hunk 5]
```diff
@@ -126,9 +129,45 @@ func (p *Provider) templateHelpers() template.FuncMap {
 		"volumeTo": func(v string) (string, error) {
 			return volumeTo(v)
 		},
+		"splitIntoLines": splitIntoLines,
 	}
 }
 
+func splitIntoLines(value interface{}) []string {
```
Comment (nightfury1204) [inline provider/k8s/template.go:136]: why this is necessary?

File: provider/k8s/template/app/service.yml.tmpl [hunk 6]
```diff
@@ -19,9 +19,13 @@ apiVersion: v1
...
-    {{ range keyValue .Annotations }}
-    {{.Key}}: "{{ quoteEscape .Value}}"
-    {{ end }}
+    {{- range $key, $value := .Annotations }}
+        {{ $key }}
+        {{- $lines := splitIntoLines $value }}
+        {{- range $lines }}
+          {{ . }}
+        {{- end }}
+    {{- end }}
```
Comment (nightfury1204) [inline provider/k8s/template/app/service.yml.tmpl:28]: why this change is needed?

File: pkg/common/format.go (hunk 1 above)
Comment (numcys) [reply pkg/common/format.go:18]: The double quotes " were being converted to html entity code &#34. so to parse that as it is in annotations I added it.

File: pkg/manifest/service.go (hunk 2 above)
Comment (numcys) [reply pkg/manifest/service.go:62]: The value could be of a string as well as map, so needed to keep this as interface to handle nested json structure.

File: pkg/manifest/service.go (hunk 3 above)
Comment (numcys) [reply pkg/manifest/service.go:366]: This is to format the json structure of the annotation properly.

File: pkg/manifest/yaml.go (hunk 4 above)
Comment (numcys) [reply pkg/manifest/yaml.go:200]: If the nested json needs to be formatted, otherwise we add the value as it is.

File: provider/k8s/template.go (hunk 5 above)
Comment (numcys) [reply provider/k8s/template.go:136]: To solve issue of templating the value of interface type, I created function to split the interface into []string, and then traverse over it and template that.

File: provider/k8s/template/app/service.yml.tmpl (hunk 6 above)
Comment (numcys) [reply provider/k8s/template/app/service.yml.tmpl:28]: This is for the above said change of printing slice of strings for a annotation instead of interface.

**Review by nightfury1204 — CHANGES_REQUESTED** [review]
I think it shouldn't be json specific implementaion. It should work for all types formated string.
